src/main/java/com/teamacronymcoders/base/subblocksystem/blocks/BlockSubBlockHolder.java

```diff
@@ -164,4 +167,75 @@ public int damageDropped(@Nonnull IBlockState state) {
     public List<IGeneratedModel> getGeneratedModels() {
         return this.getSubBlocks().values().stream().map(ISubBlock::getGeneratedModel).filter(Objects::nonNull).collect(Collectors.toList());
     }
+
+    @Override
+    public boolean isSideSolid(IBlockState blockState, @Nonnull IBlockAccess world, @Nonnull BlockPos pos, EnumFacing side) {
+        return this.getSubBlock(blockState).isSideSolid();
+    }
+
+    @Override
+    public boolean isTopSolid(IBlockState blockState) {
+        return this.getSubBlock(blockState).isTopSolid();
+    }
+
+    @Override
+    public boolean canPlaceTorchOnTop(IBlockState blockState, @Nonnull IBlockAccess world, @Nonnull BlockPos pos) {
+        return this.getSubBlock(blockState).canPlaceTorchOnTop();
+    }
+
+    @Override
+    public BlockFaceShape getBlockFaceShape(IBlockAccess world, IBlockState blockState, BlockPos pos, EnumFacing face) {
+        return this.getSubBlock(blockState).getBlockFaceShape();
+    }
+
+    @Override
+    public AxisAlignedBB getBoundingBox(IBlockState blockState, IBlockAccess source, BlockPos pos) {
+        return this.getSubBlock(blockState).getBoundingBox();
+    }
+
+    @Override
+    public boolean isFullCube(IBlockState blockState) {
+        return this.getSubBlock(blockState).isFullCube();
+    }
+
+    @Override
+    public boolean isOpaqueCube(IBlockState blockState) {
+        // blockState is null in preInit when this is first called - Used in block constructor
+        try {
+            return this.getSubBlock(blockState).isOpaqueCube();
+        } catch (NullPointerException error) {
+            return true;
+        }
+    }
+
+    @Override
+    public boolean isPassable(IBlockAccess world, BlockPos pos) {
+        return this.getSubBlock(world.getBlockState(pos)).isPassable();
+    }
+
+    @Override
+    public boolean isFullBlock(IBlockState blockState) {
+        return this.getSubBlock(blockState).isFullBlock();
+    }
+
+    @Override
+    public int getLightOpacity(IBlockState blockState)
+    {
+        return this.getSubBlock(blockState).getLightOpacity();
+    }
+
+    @Override
+    public boolean canSilkHarvest(World world, BlockPos pos, @Nonnull IBlockState blockState, EntityPlayer player) {
+        return this.getSubBlock(blockState).canSilkHarvest();
+    }
+
+    @Override
+    public void neighborChanged(IBlockState blockState, World world, BlockPos pos, Block block, BlockPos fromPos) {
+        if (!this.canPlaceBlockAt(world, pos)) {
+            if (this.getSubBlock(blockState).isBrokenWhenUnplaceable()) {
```

Comment:
shouldn't this be the first check? rather than the canPlaceBlockAt?

Reply:
I mean you can't place something where there's already a block that isn't replaceable.